pkg/server/server.go

```diff
@@ -363,11 +410,26 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 
 	ctx = typesystem.ContextWithTypesystem(ctx, typesys)
 
-	checkResolver := graph.NewLocalChecker(
+	var checkResolver graph.CheckResolver
+	localCheckResolver := graph.NewLocalChecker(
 		storagewrappers.NewCombinedTupleReader(s.datastore, req.ContextualTuples.GetTupleKeys()),
 		graph.WithResolveNodeBreadthLimit(s.resolveNodeBreadthLimit),
 		graph.WithMaxConcurrentReads(s.maxConcurrentReadsForCheck),
 	)
+	checkResolver = localCheckResolver
+
+	if s.checkCache != nil {
+		cachedCheckResolver := graph.NewCachedCheckResolver(
+			checkResolver,
```

Comment:
Tricky tricky tricky! This should be `localCheckResolver`. `checkResolver` stores a pointer reference to an implementation of the interface. When you re-assign the `checkResolver` below on line https://github.com/openfga/openfga/pull/891/files#diff-104a11712684e520f0affa95e6a053faeb713306a10f69ee99b9729cc9dd1996R431 you effecitvely set the delegate of the CachedCheckResolver to itself. You can verify this by checking that the pointer reference is the same between `checkResolver` and `cachedCheckResolver`.

* CachedCheckerResolver should delegate to LocalCheckResolver
* LocalCheckResolver should delegate to CachedCheckResolver


Right now we have:
* CachedCheckResolver delegates to CachedCheckResolver
* LocalCheckResolver delegates to CachedCheckResolver

Reply:
Hmm... we have UT that shows check works.  However, I can change it to localCheckResolver to keep things clean.